cpp/include/raft/cuda_utils.cuh

```diff
@@ -694,6 +694,41 @@ DI T blockReduce(T val, char* smem)
   return warpReduce(val);
 }
 
+/**
+ * @defgroup Load/Store Functions with cache control (for clang compilation)
+ * @{
+ */
+template <typename T>
+__device__ __forceinline__ T ldcg(const T* addr)
+{
+#if defined(__clang__)
```

Comment:
I think I found a better way of doing this, I'll revert this change.
Thanks for raising it and adding @achirkin !

Reply:
Basically, I believe what we should do here is to pre-define these functions for clang when compiling with it, and only when compiling with clang, just like clang defines the other CUDA intrinsics / global device functions (https://github.com/llvm-mirror/clang/blob/master/lib/Headers/__clang_cuda_device_functions.h).
@achirkin : given your other comment, for now, I would not actually try to ensure that clang produces anything competitive in terms of a binary. However, being able to parse the syntax of RAFT with clang allows tool like `clang-tidy` to be used, and this is really what I wanted to achieve with this PR.

Reply:
Ok, then, maybe I would add a comment for future - if someone wants to make clang compiler generate correct kernel code w.r.t. memory fences.

Reply:
Yeah, this makes sense, I'll play around with it, maybe we can use the assembly instructions directly for clang.
In any case, we won't really emit the same code with clang because optimizations are different etc.
But we can come close to it probably.